### PR TITLE
Allow skipping failure for RAT and Source Patterns

### PR DESCRIPTION
**Pull request by madrob**
I'd like to be able to still run these checks and report on them, but also be able to prevent them from failing the build. Useful in CI environment where I will have the output and also want to make sure other checks get run instead of failing fast here.

This pattern was modeled on the `validation.git.failOnModified` option.

### CODE REVIEW

File: gradle/validation/rat-sources.gradle [hunk 1]
```diff
@@ -274,8 +274,16 @@ class RatTask extends DefaultTask {
...
-                    errors.collect{ msg -> "  - ${msg}" }.join("\n"))
+            def checkProp = "validation.rat.failOnError"
+            def shouldFail = Boolean.valueOf(project.propertyOrDefault(checkProp, true))
+            def message = "Found " + errors.size() + " file(s) with errors: " +
+                    (shouldFail ? "(skip with -P${checkProp}=false)\n" : "\n") +
+                    errors.collect{ msg -> "  - ${msg}" }.join("\n")
+            if (shouldFail) {
+                throw new GradleException(message)
+            } else {
+                logger.lifecycle("NOTE: ${message}")
```
Comment (dweiss) [inline gradle/validation/rat-sources.gradle:285]: there's a logger.warning (or warn?) method you could use here instead. otherwise lgtm.

**Comment by uschindler** [comment]
Hi,
I am fine with that, but actually in an CI environment this is not needed. On Jenkins all jobs start with `gradlew --continue`, so the GradleException when thrown during task execution is converted to a warning.
So to me this seems useless as CI already works well, but it was merged already,

**Comment by uschindler** [comment]
Info: "When executed with --continue, Gradle will execute every task to be executed where all of the dependencies for that task completed without failure, instead of stopping as soon as the first failure is encountered. Each of the encountered failures will be reported at the end of the build.

If a task fails, any subsequent tasks that were depending on it will not be executed. For example, tests will not run if there is a compilation failure in the code under test; because the test task will depend on the compilation task (either directly or indirectly)."

**Comment by uschindler** [comment]
But as check tasks are last in execution (only "check" depends on it), this is fine).

**Pull request merged**